contracts/src/LightClient.sol

```diff
@@ -426,19 +426,19 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     function getHotShotCommitment(uint256 hotShotBlockHeight)
         public
         view
-        returns (BN254.ScalarField hotShotCommitment)
+        returns (HotShotCommitment memory)
     {
         uint256 commitmentsHeight = hotShotCommitments.length;
         if (hotShotBlockHeight > hotShotCommitments[commitmentsHeight - 1].blockHeight) {
             revert InvalidHotShotBlockForCommitmentCheck();
         }
         for (uint256 i = 0; i < commitmentsHeight; i++) {
             if (hotShotCommitments[i].blockHeight >= hotShotBlockHeight) {
-                return hotShotCommitments[i].blockCommRoot;
+                return hotShotCommitments[i];
             }
         }
 
-        return hotShotCommitments[commitmentsHeight - 1].blockCommRoot;
+        return hotShotCommitments[commitmentsHeight - 1];
```

Comment:
Not related to this change but isn't this unreachable? Here we would be returning something that we already visited in the loop, thus `!(hotShotCommitments[commitmentsHeight - 1].block_height >= hotShotBlockHeight)`, thus we would have reverted on line 433. I feel like it's misleading to have an unreachable `return` here and this should probably be a `revert` instead

Reply:
yep, that's right, I believe this is unreachable @alysiahuggins when you revisit this can we simplify this logic?

Reply:
I put it mainly because the compiler complained about not having a default return, would you prefer a revert instead? @nomaxg